pkg/config/replica_config.go

```diff
@@ -138,6 +139,15 @@ func (c *ReplicaConfig) Validate() error {
 	return nil
 }
 
+// Apply sinkURI to fill the `ReplicaConfig`
+func (c *ReplicaConfig) Apply(sinkURI *url.URL) *ReplicaConfig {
```

Comment:
```suggestion
func (c *ReplicaConfig) ApplyProtocol(sinkURI *url.URL) *ReplicaConfig {
```

What about naming it `ApplyProtocol`?